projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/authorization/provider/AuthorizationProviderTest.java

```diff
@@ -275,4 +276,52 @@ public void testJobGetWithTeamWithSpaces() {
 
     Assertions.assertEquals(actualBody, expectedBody);
   }
+
+  @Test
+  public void testGetAccessToken_withAuthEndpointAndRefreshToken_shouldUseObtainedAccessToken() {
+    String obtainedAccessToken = "obtainedAccessToken";
+    String receivedAccessToken = "receivedAccessToken";
+    String authorizationServerEndpoint = "authorizationServerEndpoint";
+    String refreshToken = "refreshToken";
+
+    AuthorizationProvider authorizationProvider = mock(AuthorizationProvider.class);
+    Mockito.when(
+            authorizationProvider.obtainAccessToken(
+                eq(authorizationServerEndpoint), eq(refreshToken)))
+        .thenReturn(obtainedAccessToken);
+    Mockito.when(authorizationProvider.getAccessTokenReceived()).thenReturn(receivedAccessToken);
```

Comment:
Why expecting a second call for access token retrieval?

Reply:
We don't, the test sets both methods and verifies which one is called by the returned value. It can be done by Mockito.verify () as well.